posixlib/src/main/resources/scala-native/wordexp.c

```diff
@@ -55,6 +57,14 @@ _Static_assert(offsetof(struct scalanative_wordexp_t, we_offs) ==
 #define WRDE_SYNTAX -1
 #endif // _WIN32
 
+// OpenBSD needs dummy functions as well.
+
+#if defined(__OpenBSD__)
+int wordexp(const char *restrict s, void *restrict p, int flags) { return -1; }
+
+void wordfree(void *p) {}
+#endif
```

Comment:
The tests in `org.scalanative.testsuite.posixlib.WordexpTest` should be guarded to be not run on OpenBSD

Reply:
@WojciechMazur like a few more tests, I'm in my way to run it and move forward to fix it.

Reply:
The overall "#if" handling in this and related posix C files
is getting pretty shaggy/hard_to_trace.  Fixing that should
is out-of-scope for this PR, but an Issue could be created.
The original "__unix__" testing was a complicated starting point
and it has gotten more complicated since. Having a code formatting style which does not allow indented "#" blocks
also makes things harder to follow.

What we can do is to try to propagate the OpenBSD changes
to other files in as traceable manner as feasible.

A suggestion.  The lines 60-66 could be moved to just after
line 57 and the comment on current line 58 corrected.

To my thinking/tracing once my mind has gone into a 
"if _WIN32 or OpenBSD" block it wants to stay with those
operating systems rather that pop out at an "endif" and then
go back into one of those same operating systems.



Should be functionally the same.